nixops/backends/ec2.py

```diff
@@ -871,10 +884,13 @@ def create(self, defn, check, allow_reboot, allow_recreate):
             # devices defined in the spec.  These cannot be changed
             # later.
             devmap = boto.ec2.blockdevicemapping.BlockDeviceMapping()
-            devs_mapped = {}
+
             for k, v in defn.block_device_mapping.iteritems():
-                if re.match("/dev/sd[a-e]", k) and not v['disk'].startswith("ephemeral"):
-                    raise Exception("non-ephemeral disk not allowed on device ‘{0}’; use /dev/xvdf or higher".format(_sd_to_xvd(k)))
+                # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
+                device_name_recommended_for_ebs_volumes = re.match("/dev/sd[a-e]", k) or re.match("/dev/xvd[a-e]", k) or re.match("/dev/nvme0n1", k)
```

Comment:
misname again, should be `device_names_not_recommended` instead :)

Reply:
oups

Reply:
actually think that `or re.match("/dev/nvme0n1", k)` case is not possible, I'll better remove it from list